src/raml.coffee

```diff
@@ -171,14 +172,39 @@ class @RamlParser
         return if lastVisitedNode then lastVisitedNode else node
       )
 
-  getPendingFile: (loader, fileInfo) ->
+  dereferenceSchema: (file) ->
+    deferred = @q.defer()
+    schemaLoader.dereference(file)
+      .then((schema) ->
+        deferred.resolve(JSON.stringify(schema))
+      ).catch((err) ->
+        deferred.reject(err)
+      )
+    return deferred.promise
+
+  checkPendingFileIsSchema: (loader, fileInfo, node) ->
+    if fileInfo.parentKey and fileInfo.parentKey.value is 'schema'
+      return true
+
+    if fileInfo.parentNode
+      schemasSeqNode = loader.findSchemasSequenceNode(node)
```

Comment:
@dmartinezg can you please give an example how `someSlowFile.yaml` should look like?